pkg/controllers/nodeclaim/lifecycle/liveness.go

```diff
@@ -42,12 +42,22 @@ type Liveness struct {
 // registrationTTL is a heuristic time that we expect the node to register within
 // If we don't see the node within this time, then we should delete the NodeClaim and try again
 const registrationTTL = time.Minute * 15
+const launchTTL = time.Minute * 5
 
 func (l *Liveness) Reconcile(ctx context.Context, nodeClaim *v1.NodeClaim) (reconcile.Result, error) {
 	registered := nodeClaim.StatusConditions().Get(v1.ConditionTypeRegistered)
 	if registered.IsTrue() {
 		return reconcile.Result{}, nil
 	}
+	launched := nodeClaim.StatusConditions().Get(v1.ConditionTypeLaunched)
+	if launched == nil {
+		return reconcile.Result{Requeue: true}, nil
+	}
+	if ttl := launchTTL - l.clock.Since(launched.LastTransitionTime.Time); ttl <= 0 {
+		if err := l.deleteNodeClaimForTTL(ctx, "launch_ttl", nodeClaim); err != nil {
```

Comment:
```suggestion
		if err := l.deleteNodeClaimForTTL(ctx, "launch_timeout", nodeClaim); err != nil {
```

I think I prefer using the term timeout here over `ttl`